src/plugins/intel_cpu/src/nodes/kernels/grid_sample_kernel.cpp

```diff
@@ -0,0 +1,1572 @@
+        uni_vfnmadd231ps(vCoordDst, vAux, ptr[regAux4]); // abs(x) % D21
+    } else {
+        // (x % D2 + D2) % D2
+        if (vCoordDst.getIdx() != vCoordOrigin.getIdx())
+            uni_vmovups(vCoordDst, vCoordOrigin);
+        uni_vmovups(vAux, vCoordOrigin);
+        if (dim == coord::w) {
+            mov(regAux4, ptr[regParams + GET_OFF(srcWidthMul2F)]);
+        } else if (coord::h) {
+            mov(regAux4, ptr[regParams + GET_OFF(srcHeightMul2F)]);
+        }
+        uni_vdivps(vAux, vAux, ptr[regAux4]);
+        uni_vroundps(vAux, vAux, 0x3); // Round floor
+        uni_vfnmadd231ps(vCoordDst, vAux, ptr[regAux4]); // x % D2
+        uni_vaddps(vCoordDst, vCoordDst, ptr[regAux4]); // x % D2 + D2
+        uni_vmovups(vAux, vCoordDst);
+        uni_vdivps(vAux, vAux, ptr[regAux4]);
+        uni_vroundps(vAux, vAux, 0x3); // Round floor
+        uni_vfnmadd231ps(vCoordDst, vAux, ptr[regAux4]); // (x % D2 + D2) % D2
+    }
+
+    if (dim == coord::w) {
+        mov(regAux4, ptr[regParams + GET_OFF(srcWidthMul2Sub1F)]);
+    } else if (coord::h) {
+        mov(regAux4, ptr[regParams + GET_OFF(srcHeightMul2Sub1F)]);
+    }
+    uni_vmovups(vAux, ptr[regAux4]);
+    uni_vsubps(vAux, vAux, vCoordDst);
+    if (dim == coord::w) {
+        uni_vmovups(kAux, vSrcWidthF);
+    } else if (coord::h) {
+        uni_vmovups(kAux, vSrcHeightF);
+    }
+    vcmpps(kAux, kAux, vCoordDst, 0x2); // vCoordDst >= vSrcDimF
+    uni_vaddps(vCoordDst, vCoordDst, kAux);
+    uni_vaddps(kAux, kAux, vAux);
+    uni_vaddps(vCoordDst, vCoordDst, kAux);
+}
+
+template <>
+void jitGridSampleKernel<x64::avx512_core>::bicubicCoefficients(const Vmm& vCoef, const Vmm& vDDim, const Vmm* vAuxPool, const uint8_t idx) {
+    if (idx == 0) {
+        uni_vmovups(vCoef, vDDim);
+        vfnmadd132ps(vCoef, vOnesF, v2val);
+        uni_vfmadd231ps(vCoef, vDDim, vDDim);
+        uni_vmulps(vCoef, vCoef, vDDim);
+        uni_vmulps(vCoef, vCoef, vBicubConst);
+    } else if (idx == 1) {
+        uni_vmovups(vCoef, vDDim);
+        vfmsub132ps(vCoef, vBicub3Const, vBicub2Const);
+        uni_vmulps(vCoef, vCoef, vDDim);
+        uni_vfmadd132ps(vCoef, vOnesF, vDDim);
+    } else if (idx == 2) {
+        uni_vmulps(vCoef, vDDim, vDDim);
+        vfmadd132ps(vCoef, vBicubConst, vBicub2Const);
+        vfmsub231ps(vCoef, vDDim, v2Bicub3Const);
+        uni_vmulps(vCoef, vCoef, vDDim);
+    } else if (idx == 3) {
+        uni_vmulps(vCoef, vBicubConst, vDDim);
+        uni_vmulps(vCoef, vCoef, vDDim);
+        vfnmadd132ps(vCoef, vCoef, vDDim);
+    }
+}
+
+template <>
+void jitGridSampleKernel<x64::sse41>::bicubicCoefficients(const Vmm& vCoef, const Vmm& vDDim, const Vmm* vAuxPool, const uint8_t idx) {
+    const auto& vAux = vAuxPool[0];
+    if (idx == 0) {
+        static const float values2[4] = { 2.f, 2.f, 2.f, 2.f };
+        mov(regAux1, reinterpret_cast<uintptr_t>(values2));
+        uni_vmovups(vCoef, vDDim);
+        vfnmadd132ps(vCoef, vOnesF, v2val);
```

Comment:
I'm not really sure that it will work on SSE4.1.... Could you please verify?

The same comment for below lines with FMA instructions without `uni` prefix

Reply:
fixed